invenio/modules/knowledge/models.py

```diff
@@ -166,6 +185,46 @@ def set_dyn_config(self, field, expression, collection=None):
                                     search_expression=expression,
                                     collection=collection)
 
+    @staticmethod
+    def generate_slug(name):
+        """Generate a slug for the knowledge.
+
+        :param name: text to slugify
+        :return: slugified text
+        """
+        slug = slugify(unicode(name))
```

Comment:
@jirikuncar I move the conversion in unicode here (the same in the upgrade script). :)